### PR TITLE
Add SSL root certificate support

### PR DESCRIPTION
**Pull request by comphead**

### CODE REVIEW

**Comment by comphead** [comment]
@nooberfsh please check the PR

**Comment by nooberfsh** [comment]
> @nooberfsh please check the PR

Hi, thanks for the PR, I will review it after work.

File: src/ssl.rs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+use std::fmt;
+use std::io::Read;
+
+#[derive(Clone)]
+
+pub struct Ssl {
+    pub enabled: bool,
+    pub root_cert: Option<CertificateType>,
+}
+
```
Comment (nooberfsh) [inline src/ssl.rs:11]: Can you elaborate Why this type is necessary?

File: src/ssl.rs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+///        .auth(Auth::Basic("user".to_string(), Some("pwd".to_string())))
+///        .secure(true)
+///        .ssl(Ssl {
+///            enabled: true,
+///            root_cert: Some(CertificateType::Pem("/path/root.pem".to_string()))
+///        })
+///        .catalog("catalog")
+///        .build()
+///        .unwrap();
+*/
```
Comment (nooberfsh) [inline src/ssl.rs:30]: Does rust having this `/*...*/`  styling?

File: src/ssl.rs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            root_cert: None,
+        }
+    }
+}
+
+#[derive(Clone)]
+pub enum CertificateType {
+    Pem(String),
+    Der(String),
+}
```
Comment (nooberfsh) [inline src/ssl.rs:44]: What about this:
```rust
pub struct Certificate(reqwest::Certificate);
```

File: src/ssl.rs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+impl CertificateType {
+    pub fn new_pem(root_certificate_path: &str) -> Result<reqwest::Certificate> {
```
Comment (nooberfsh) [inline src/ssl.rs:47]: I think the path param can be represented as
```rust
pub fn new_pem<P: AsRef<Path>>(root_certificate_path:  P) -> {...}
```

File: src/ssl.rs [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        let mut buf = Vec::new();
+        std::fs::File::open(root_certificate_path)
+            .expect(&format!(
+                "Error opening certificate file {}",
+                root_certificate_path
+            ))
```
Comment (nooberfsh) [inline src/ssl.rs:53]: I don't think `expect` is right here, we should return the error to the caller.

**Pull request closed without merging**